<html><head><title>Extract significant genes for sets of variables in  time series gene expression experiments</title>
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>get.siggenes(maSigPro)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   get.siggenes">
<param name="keyword" value=" Extract significant genes for sets of variables in  time series gene expression experiments">
</object>


<h2>Extract significant genes for sets of variables in  time series gene expression experiments</h2>


<h3>Description</h3>

<p>
This function creates lists of significant genes for a set of variables whose significance value has been computed with the <code>T.fit</code> function.
</p>


<h3>Usage</h3>

<pre>
get.siggenes(tstep, rsq = 0.7, add.IDs = FALSE, IDs = NULL, matchID.col = 1, 
             only.names = FALSE, vars = c("all", "each", "groups"), 
             groups.vector = NULL, trat.repl.spots = "none", 
             index = IDs[, (matchID.col + 1)], match = IDs[, matchID.col], r = 0.7)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>tstep</code></td>
<td>
a <code>T.fit</code> object </td></tr>
<tr valign="top"><td><code>rsq</code></td>
<td>
cut-off level at the R-squared value for the stepwise regression fit. Only genes with R-squared more than rsq are selected</td></tr>
<tr valign="top"><td><code>add.IDs</code></td>
<td>
logical indicating whether to include additional gene id's in the result</td></tr>
<tr valign="top"><td><code>IDs</code></td>
<td>
matrix contaning additional gene id information (required when <code>add.IDs</code> is <code>TRUE</code>) </td></tr>
<tr valign="top"><td><code>matchID.col</code></td>
<td>
number of matching column in matrix IDs for adding genes ids </td></tr>
<tr valign="top"><td><code>only.names</code></td>
<td>
logical. If <code>TRUE</code>, expression values are ommited in the results </td></tr>
<tr valign="top"><td><code>vars</code></td>
<td>
variables for which to extract significant genes (see details) </td></tr>
<tr valign="top"><td><code>groups.vector</code></td>
<td>
required when <code>vars</code> is <code>"groups"</code>.</td></tr>
<tr valign="top"><td><code>trat.repl.spots</code></td>
<td>
treatment given to replicate spots. Possible values are <code>"none"</code> and <code>"average"</code></td></tr>
<tr valign="top"><td><code>index</code></td>
<td>
argument of the <code><a href="average.rows.html">average.rows</a></code> function to use when <code> trat.repl.spots</code> is <code>"average"</code> </td></tr>
<tr valign="top"><td><code>match</code></td>
<td>
argument of the <code><a href="average.rows.html">average.rows</a></code> function to use when <code>trat.repl.spots</code> is <code>"average"</code> </td></tr>
<tr valign="top"><td><code>r</code></td>
<td>
minimun pearson correlation coefficient for replicated spots profiles to be averaged </td></tr>
</table>

<h3>Details</h3>

<p>
There are 3 possible values for the vars argument:
newline qquad <code>"all"</code>: generates one single matrix or gene list with all significant genes.
newline qquad <code>"each"</code>: generates as many significant genes extractions as variables in the general regression model. Each extraction contains the significant genes for that variable.
newline qquad <code>"groups"</code>: generates a significant genes extraction for each experimental group.  
newline  The difference between <code>"each"</code> and <code>"groups"</code> is that in the first case the variables of the same group (e.g.  <code>"TreatmentA"</code> and 
<code>"time*TreatmentA" </code>) will be extracted separately and in the second case jointly.
</p>
<p>
When <code>add.IDs</code> is <code>TRUE</code>, a matrix of gene ids must be provided as argument of IDs, the <code>matchID.col</code> 
column of which having same levels as in the row names of <code>sig.profiles</code>. The option <code>only.names</code> is <code>TRUE</code> will generate a vector of significant genes or a matrix when <code>add.IDs</code> is set also to <code>TRUE</code>.
</p>
<p>
When <code>trat.repl.spots</code> is <code>"average"</code>, <code>match</code> and <code>index</code> vectors are required for the <code><a href="average.rows.html">average.rows</a></code> function.
In gene expression data context, the <code>index</code> vector would contain geneIDs and indicate which spots 
are replicates. The <code>match</code> vector is used to match these genesIDs to rows in the significant genes 
matrix, and must have the same levels as the row names of <code>sig.profiles</code>.
</p>
<p>
<code>add.IDs</code> = TRUE and <code>trat.repl.spots</code> = <code>"average"</code> are not compatible argumet values.
<code>add.IDs</code> = TRUE and <code>only.names</code> = <code>TRUE</code> are  compatible argumet values.
</p>


<h3>Value</h3>

<table summary="R argblock">
<tr valign="top"><td><code>summary</code></td>
<td>
a vector or matrix listing significant genes for the variables given by the function parameters</td></tr>
<tr valign="top"><td><code>sig.genes</code></td>
<td>
a list with detailed information on the significant genes found for the variables given by the function parameters. Each element of the list is also a list containing:
newline quad <code>sig.profiles</code>: expression values of significant genes
newline quad <code>coefficients</code>: regression coefficients for significant genes
newline quad <code>t.score</code>: t.score of significant genes
newline quad <code>sig.pvalues</code>: p-values of the regression coefficients for significant genes
newline quad <code>g</code>: number of genes
newline quad <code>...</code>: arguments passed by previous functions</td></tr>
</table>

<h3>Author(s)</h3>

<p>
Ana Conesa, aconesa@ivia.es; María José Nueda, mj.nueda@ua.es
</p>


<h3>References</h3>

<p>
Conesa, A., Nueda M.J., Alberto Ferrer, A., Talón, T. 2005.
maSigPro: a Method to Identify Significant Differential Expression Profiles in Time-Course Microarray Experiments.
</p>


<h3>Examples</h3>

<pre>

#### GENERATE TIME COURSE DATA
## generate n random gene expression profiles of a data set with 
## one control plus 3 treatments, 3 time points and r replicates per time point.

tc.GENE &lt;- function(n, r,
             var11 = 0.01, var12 = 0.01,var13 = 0.01,
             var21 = 0.01, var22 = 0.01, var23 =0.01,
             var31 = 0.01, var32 = 0.01, var33 = 0.01,
             var41 = 0.01, var42 = 0.01, var43 = 0.01,
             a1 = 0, a2 = 0, a3 = 0, a4 = 0,
             b1 = 0, b2 = 0, b3 = 0, b4 = 0,
             c1 = 0, c2 = 0, c3 = 0, c4 = 0)
{

  tc.dat &lt;- NULL
  for (i in 1:n) {
    Ctl &lt;- c(rnorm(r, a1, var11), rnorm(r, b1, var12), rnorm(r, c1, var13))  # Ctl group
    Tr1 &lt;- c(rnorm(r, a2, var21), rnorm(r, b2, var22), rnorm(r, c2, var23))  # Tr1 group
    Tr2 &lt;- c(rnorm(r, a3, var31), rnorm(r, b3, var32), rnorm(r, c3, var33))  # Tr2 group
    Tr3 &lt;- c(rnorm(r, a4, var41), rnorm(r, b4, var42), rnorm(r, c4, var43))  # Tr3 group
    gene &lt;- c(Ctl, Tr1, Tr2, Tr3)
    tc.dat &lt;- rbind(tc.dat, gene)
  }
  tc.dat
}
## Create 270 flat profiles
flat &lt;- tc.GENE(n = 270, r = 3)
## Create 10 genes with profile differences between Ctl and Tr1 groups
twodiff &lt;- tc.GENE (n = 10, r = 3, b2 = 0.5, c2 = 1.3)
## Create 10 genes with profile differences between Ctl, Tr2, and Tr3 groups
threediff &lt;- tc.GENE(n = 10, r = 3, b3 = 0.8, c3 = -1, a4 = -0.1, b4 = -0.8, c4 = -1.2)
## Create 10 genes with profile differences between Ctl and Tr2 and different variance
vardiff &lt;- tc.GENE(n = 10, r = 3, a3 = 0.7, b3 = 1, c3 = 1.2, var32 = 0.03, var33 = 0.03)
## Create dataset
tc.DATA &lt;- rbind(flat, twodiff, threediff, vardiff)
rownames(tc.DATA) &lt;- paste("feature", c(1:300), sep = "")
colnames(tc.DATA) &lt;- paste("Array", c(1:36), sep = "")
tc.DATA [sample(c(1:(300*36)), 300)] &lt;- NA  # introduce missing values

#### CREATE EXPERIMENTAL DESIGN
Time &lt;- rep(c(rep(c(1:3), each = 3)), 4)
Replicates &lt;- rep(c(1:12), each = 3)
Control &lt;- c(rep(1, 9), rep(0, 27))
Treat1 &lt;- c(rep(0, 9), rep(1, 9), rep(0, 18))
Treat2 &lt;- c(rep(0, 18), rep(1, 9), rep(0,9))
Treat3 &lt;- c(rep(0, 27), rep(1, 9))
edesign &lt;- cbind(Time, Replicates, Control, Treat1, Treat2, Treat3)
rownames(edesign) &lt;- paste("Array", c(1:36), sep = "")

tc.p &lt;- p.vector(tc.DATA, design = make.design.matrix(edesign), Q = 0.01) 
tc.tstep &lt;- T.fit(data = tc.p , alfa = 0.05)

## This will obtain sigificant genes per experimental group 
## which have a regression model Rsquared &gt; 0.9
tc.sigs &lt;- get.siggenes (tc.tstep, rsq = 0.9, vars = "groups")

## This will obtain all sigificant genes regardless the Rsquared value. 
## Replicated genes are averaged.
IDs &lt;- rbind(paste("feature", c(1:300), sep = ""), 
       rep(paste("gene", c(1:150), sep = ""), each = 2))
tc.sigs.ALL &lt;- get.siggenes (tc.tstep, rsq = 0, vars = "all", IDs = IDs)

</pre>



<hr><div align="center">[Package <em>maSigPro</em> version 1.0 <a href="00Index.html">Index]</a></div>

</body></html>
