<html><head><title>Make regression fit for time series gene expression experiments</title>
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>p.vector(maSigPro)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   p.vector">
<param name="keyword" value=" Make regression fit for time series gene expression experiments">
</object>


<h2>Make regression fit for time series gene expression experiments</h2>


<h3>Description</h3>

<p>
<code>p.vector</code> performs a regression fit for each gene taking all variables present in the model given by a regression matrix
and returns a list of FDR corrected significant genes.
</p>


<h3>Usage</h3>

<pre>
p.vector(data, design = NULL, Q = 0.05, fdr.method = "BH", min.obs = 3)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>data</code></td>
<td>
matrix containing normalized gene expression data. Genes must be in rows and arrays in columns</td></tr>
<tr valign="top"><td><code>design</code></td>
<td>
design matrix for the regression fit such as that generated by the <code><a href="make.design.matrix.html">make.design.matrix</a></code> function</td></tr>
<tr valign="top"><td><code>Q</code></td>
<td>
level of false discovery rate (FDR) control</td></tr>
<tr valign="top"><td><code>fdr.method</code></td>
<td>
method for coputing FDR. Currently only the Benajamini &amp; Holderberg method is available</td></tr>
<tr valign="top"><td><code>min.obs</code></td>
<td>
genes with less than this number of true numerical values will be excluded from the analysis. Default is  3 (minimun value for a quadratic fit) </td></tr>
</table>

<h3>Details</h3>

<p>
<code>rownames(design)</code> and <code>colnames(data)</code> must be identical vectors and indicate array naming.
newline   <code>rownames(data)</code> should contain unique gene IDs.
newline   <code>colnames(design)</code> are the given names for the variables in the regression model.
</p>


<h3>Value</h3>

<table summary="R argblock">
<tr valign="top"><td><code>SELEC</code></td>
<td>
matrix containing the expression values for significant genes</td></tr>
<tr valign="top"><td><code>p.vector </code></td>
<td>
vector containing the computed p-values</td></tr>
<tr valign="top"><td><code>G </code></td>
<td>
total number of input genes</td></tr>
<tr valign="top"><td><code>g</code></td>
<td>
number of genes taken in the regression fit</td></tr>
<tr valign="top"><td><code>alfa </code></td>
<td>
p-value at FDR <code> Q</code> control</td></tr>
<tr valign="top"><td><code>i</code></td>
<td>
number of significant genes</td></tr>
<tr valign="top"><td><code>dis </code></td>
<td>
design matrix used in the regression fit</td></tr>
<tr valign="top"><td><code>dat </code></td>
<td>
matrix of expression value data used in the regression fit</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
additional values from input parameters</td></tr>
</table>

<h3>Author(s)</h3>

<p>
Ana Conesa, aconesa@ivia.es; María José Nueda, mj.nueda@ua.es
</p>


<h3>References</h3>

<p>
Conesa, A., Nueda M.J., Alberto Ferrer, A., Talón, T. 2005.
maSigPro: a Method to Identify Significant Differential Expression Profiles in Time-Course Microarray Experiments.
</p>


<h3>See Also</h3>

<p>
<code><a href="T.fit.html">T.fit</a></code>, <code><a onclick="findlink('stats', 'lm.html')" style="text-decoration: underline; color: blue; cursor: hand">lm</a></code>
</p>


<h3>Examples</h3>

<pre>
#### GENERATE TIME COURSE DATA
## generates n random gene expression profiles of a data set with 
## one control plus 3 treatments, 3 time points and r replicates per time point.

tc.GENE &lt;- function(n, r,
             var11 = 0.01, var12 = 0.01,var13 = 0.01,
             var21 = 0.01, var22 = 0.01, var23 =0.01,
             var31 = 0.01, var32 = 0.01, var33 = 0.01,
             var41 = 0.01, var42 = 0.01, var43 = 0.01,
             a1 = 0, a2 = 0, a3 = 0, a4 = 0,
             b1 = 0, b2 = 0, b3 = 0, b4 = 0,
             c1 = 0, c2 = 0, c3 = 0, c4 = 0)
{

  tc.dat &lt;- NULL
  for (i in 1:n) {
    Ctl &lt;- c(rnorm(r, a1, var11), rnorm(r, b1, var12), rnorm(r, c1, var13))  # Ctl group
    Tr1 &lt;- c(rnorm(r, a2, var21), rnorm(r, b2, var22), rnorm(r, c2, var23))  # Tr1 group
    Tr2 &lt;- c(rnorm(r, a3, var31), rnorm(r, b3, var32), rnorm(r, c3, var33))  # Tr2 group
    Tr3 &lt;- c(rnorm(r, a4, var41), rnorm(r, b4, var42), rnorm(r, c4, var43))  # Tr3 group
    gene &lt;- c(Ctl, Tr1, Tr2, Tr3)
    tc.dat &lt;- rbind(tc.dat, gene)
  }
  tc.dat
}

## Create 270 flat profiles
flat &lt;- tc.GENE(n = 270, r = 3)
## Create 10 genes with profile differences between Ctl and Tr1 groups
twodiff &lt;- tc.GENE (n = 10, r = 3, b2 = 0.5, c2 = 1.3)
## Create 10 genes with profile differences between Ctl, Tr2, and Tr3 groups
threediff &lt;- tc.GENE(n = 10, r = 3, b3 = 0.8, c3 = -1, a4 = -0.1, b4 = -0.8, c4 = -1.2)
## Create 10 genes with profile differences between Ctl and Tr2 and different variance
vardiff &lt;- tc.GENE(n = 10, r = 3, a3 = 0.7, b3 = 1, c2 = 1.3, var32 = 0.03, var33 = 0.03)
## Create dataset
tc.DATA &lt;- rbind(flat, twodiff, threediff, vardiff)
rownames(tc.DATA) &lt;- paste("feature", c(1:300), sep = "")
colnames(tc.DATA) &lt;- paste("Array", c(1:36), sep = "")
tc.DATA [sample(c(1:(300*36)), 300)] &lt;- NA  # introduce missing values

#### CREATE EXPERIMENTAL DESIGN
Time &lt;- rep(c(rep(c(1:3), each = 3)), 4)
Replicates &lt;- rep(c(1:12), each = 3)
Control &lt;- c(rep(1, 9), rep(0, 27))
Treat1 &lt;- c(rep(0, 9), rep(1, 9), rep(0, 18))
Treat2 &lt;- c(rep(0, 18), rep(1, 9), rep(0,9))
Treat3 &lt;- c(rep(0, 27), rep(1, 9))
edesign &lt;- cbind(Time, Replicates, Control, Treat1, Treat2, Treat3)
rownames(edesign) &lt;- paste("Array", c(1:36), sep = "")

tc.p &lt;- p.vector(tc.DATA, design = make.design.matrix(edesign), Q = 0.01)
tc.p$i # number of significant genes
tc.p$SELEC # expression value of signficant genes
tc.p$alfa # p.value at FDR control

</pre>

<script Language="JScript">
function findlink(pkg, fn) {
var Y, link;
Y = location.href.lastIndexOf("\\") + 1;
link = location.href.substring(0, Y);
link = link + "../../" + pkg + "/chtml/" + pkg + ".chm::/" + fn;
location.href = link;
}
</script>


<hr><div align="center">[Package <em>maSigPro</em> version 1.0 <a href="00Index.html">Index]</a></div>

</body></html>
