<html><head><title>Function for plotting gene expression profile at different experimental groups</title>
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>PlotGroups(maSigPro)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   PlotGroups">
<param name="keyword" value=" Function for plotting gene expression profile at different experimental groups">
</object>


<h2>Function for plotting gene expression profile at different experimental groups</h2>


<h3>Description</h3>

<p>
This function displays the gene expression profile for each experimental group in a time series
gene expression experiment.
</p>


<h3>Usage</h3>

<pre>
PlotGroups(data, edesign = NULL, time = edesign[,1], groups = edesign[,c(3:ncol(edesign))], 
           repvect = edesign[,2], show.fit = FALSE, dis = NULL, step.method = "backward", 
           min.obs = 2, alfa = 0.01, show.lines = TRUE, groups.vector = NULL, xlab = "time", cex.xaxis = 1, 
           ylim = NULL, main = NULL, cexlab = 0.8, legend = TRUE, sub = NULL)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>data</code></td>
<td>
vector or matrix containing the gene expression data </td></tr>
<tr valign="top"><td><code>edesign</code></td>
<td>
matrix describing experimental design. Rows must be arrays and columns experiment descriptors</td></tr>
<tr valign="top"><td><code>time</code></td>
<td>
vector indicating time assigment for each array </td></tr>
<tr valign="top"><td><code>groups</code></td>
<td>
matrix indicating experimental group to which each array is assigned </td></tr>
<tr valign="top"><td><code>repvect</code></td>
<td>
index vector indicating experimental replicates</td></tr>
<tr valign="top"><td><code>show.fit</code></td>
<td>
logical indicating whether regression fit curves must be plotted</td></tr>
<tr valign="top"><td><code>dis</code></td>
<td>
regression design matrix  </td></tr>
<tr valign="top"><td><code>step.method</code></td>
<td>
stepwise regression method to fit models for cluster mean profiles. It can be either <code>"backward"</code>, <code>"forward"</code>, <code>"two.ways.backward"</code> or <code>"two.ways.forward"</code> </td></tr>
<tr valign="top"><td><code>min.obs</code></td>
<td>
minimal number of observations for a gene to be included in the analysis</td></tr>
<tr valign="top"><td><code>alfa</code></td>
<td>
significance level used for variable selection in the stepwise regression </td></tr>
<tr valign="top"><td><code>show.lines</code></td>
<td>
logical indicating whether a line must be drawn joining plotted data points for reach group</td></tr>
<tr valign="top"><td><code>groups.vector</code></td>
<td>
vector indicating experimental group to which each variable belongs </td></tr>
<tr valign="top"><td><code>xlab</code></td>
<td>
label for the x axis </td></tr>
<tr valign="top"><td><code>cex.xaxis</code></td>
<td>
graphical parameter maginfication to be used for x axis in plotting functions </td></tr>
<tr valign="top"><td><code>ylim</code></td>
<td>
range of the y axis </td></tr>
<tr valign="top"><td><code>main</code></td>
<td>
plot main title </td></tr>
<tr valign="top"><td><code>cexlab</code></td>
<td>
graphical parameter maginfication to be used for x axis label in plotting functions  </td></tr>
<tr valign="top"><td><code>legend</code></td>
<td>
logical indicating whether legend must be added when plotting profiles</td></tr>
<tr valign="top"><td><code>sub</code></td>
<td>
plot subtitle </td></tr>
</table>

<h3>Details</h3>

<p>
To compute experimental groups either a edesign object must be provided, or separate values must be given for the <code>time</code>, <code>repvect</code> and <code>groups</code> 
arguments.
newline
When data is a matrix, the average expression value is displayed.
newline
When there are array replicates in the data (as indicated by <code>repvect</code>), values are averaged by <code>repvect</code>.
newline
When <code>show.fit</code> is <code>TRUE</code> the stepwise regression fit for the data will be computed and the regression curves will be displayed. 
If data is a matrix of genes, the regression fit will be computed for the averaged expression value.
</p>


<h3>Value</h3>

<p>
Plot of gene expression profiles by-group.</p>

<h3>Author(s)</h3>

<p>
Ana Conesa, aconesa@ivia.es; María José Nueda, mj.nueda@ua.es
</p>


<h3>References</h3>

<p>
Conesa, A., Nueda M.J., Alberto Ferrer, A., Talón, T. 2005.
maSigPro: a Method to Identify Significant Differential Expression Profiles in Time-Course Microarray Experiments.
</p>


<h3>See Also</h3>

<p>
<code><a href="PlotProfiles.html">PlotProfiles</a></code>
</p>


<h3>Examples</h3>

<pre>

#### GENERATE TIME COURSE DATA
## generate n random gene expression profiles of a data set with 
## one control plus 3 treatments, 3 time points and r replicates per time point.

tc.GENE &lt;- function(n, r,
             var11 = 0.01, var12 = 0.01,var13 = 0.01,
             var21 = 0.01, var22 = 0.01, var23 =0.01,
             var31 = 0.01, var32 = 0.01, var33 = 0.01,
             var41 = 0.01, var42 = 0.01, var43 = 0.01,
             a1 = 0, a2 = 0, a3 = 0, a4 = 0,
             b1 = 0, b2 = 0, b3 = 0, b4 = 0,
             c1 = 0, c2 = 0, c3 = 0, c4 = 0)
{

  tc.dat &lt;- NULL
  for (i in 1:n) {
    Ctl &lt;- c(rnorm(r, a1, var11), rnorm(r, b1, var12), rnorm(r, c1, var13))  # Ctl group
    Tr1 &lt;- c(rnorm(r, a2, var21), rnorm(r, b2, var22), rnorm(r, c2, var23))  # Tr1 group
    Tr2 &lt;- c(rnorm(r, a3, var31), rnorm(r, b3, var32), rnorm(r, c3, var33))  # Tr2 group
    Tr3 &lt;- c(rnorm(r, a4, var41), rnorm(r, b4, var42), rnorm(r, c4, var43))  # Tr3 group
    gene &lt;- c(Ctl, Tr1, Tr2, Tr3)
    tc.dat &lt;- rbind(tc.dat, gene)
  }
  tc.dat
}

## create 10 genes with profile differences between Ctl, Tr2, and Tr3 groups
tc.DATA &lt;- tc.GENE(n = 10,r = 3, b3 = 0.8, c3 = -1, a4 = -0.1, b4 = -0.8, c4 = -1.2)
rownames(tc.DATA) &lt;- paste("gene", c(1:10), sep = "")
colnames(tc.DATA) &lt;- paste("Array", c(1:36), sep = "")

#### CREATE EXPERIMENTAL DESIGN
Time &lt;- rep(c(rep(c(1:3), each = 3)), 4)
Replicates &lt;- rep(c(1:12), each = 3)
Ctl &lt;- c(rep(1, 9), rep(0, 27))
Tr1 &lt;- c(rep(0, 9), rep(1, 9), rep(0, 18))
Tr2 &lt;- c(rep(0, 18), rep(1, 9), rep(0, 9))
Tr3 &lt;- c(rep(0, 27), rep(1, 9))

PlotGroups (tc.DATA, time = Time, repvect = Replicates, groups = cbind(Ctl, Tr1, Tr2, Tr3))

</pre>



<hr><div align="center">[Package <em>maSigPro</em> version 1.0 <a href="00Index.html">Index]</a></div>

</body></html>
