<html><head><title>Makes a stepwise regression fit for time series gene expression experiments</title>
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>T.fit(maSigPro)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   T.fit">
<param name="keyword" value=" Makes a stepwise regression fit for time series gene expression experiments">
</object>


<h2>Makes a stepwise regression fit for time series gene expression experiments</h2>


<h3>Description</h3>

<p>
<code>T.fit</code> selects the best regression model for each gene using stepwise regression.
</p>


<h3>Usage</h3>

<pre>
T.fit(data, design = data$dis, step.method = "backward", 
      min.obs = data$min.obs, alfa = data$Q)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>data</code></td>
<td>
can either be a <code><a href="p.vector.html">p.vector</a></code> object or a matrix containing expression data with the same requirements as for
the <code><a href="p.vector.html">p.vector</a></code> function</td></tr>
<tr valign="top"><td><code>design</code></td>
<td>
design matrix for the regression fit such as that generated by the <code><a href="make.design.matrix.html">make.design.matrix</a></code> function. If data is
a <code><a href="p.vector.html">p.vector</a></code> object, the same design matrix is used by default </td></tr>
<tr valign="top"><td><code>step.method</code></td>
<td>
argument to be passed to the step function. Can be either <code>"backward"</code>, <code>"forward"</code>, <code>"two.ways.backward"</code> or <code>"two.ways.forward"</code> </td></tr>
<tr valign="top"><td><code>min.obs</code></td>
<td>
genes with less than this number of true numerical values will be excluded from the analysis </td></tr>
<tr valign="top"><td><code>alfa</code></td>
<td>
significance level used for variable selection in the stepwise regression </td></tr>
</table>

<h3>Details</h3>

<p>
In the maSigPro approach <code><a href="p.vector.html">p.vector</a></code> and <code><a href="T.fit.html">T.fit</a></code> are subsequent steps, meaning that significant genes are
first selected on the basis of a general model and then the significant variables for each gene are found by step-wise regression. 
</p>
<p>
The step regression can be <code>"backward"</code> or <code>"forward"</code> indicating whether the step procedure starts from the 
model with all or none variables. With the <code>"two.ways.backward"</code> or <code>"two.ways.forward"</code> options the variables are both allowed to get in and out.
At each step the p-value of each variable is computed and variables get in/out the model when this p-value is
lower or higher than given threshold alfa.
</p>


<h3>Value</h3>

<table summary="R argblock">
<tr valign="top"><td><code>sol</code></td>
<td>
matrix for summary results of the stepwise regression. For each selected gene the following values are given:
newline qquad p-value of the regression ANOVA, 
newline qquad R-squared of the model 
newline qquad p-value of the regression coefficients of the selected variables</td></tr>
<tr valign="top"><td><code>sig.profiles</code></td>
<td>
expression values for the genes contained in <code>sol</code></td></tr>
<tr valign="top"><td><code>coefficients </code></td>
<td>
matrix containing regression coefficients for the adjusted models</td></tr>
<tr valign="top"><td><code>t.score </code></td>
<td>
matix containing t.scores for the adjusted models</td></tr>
<tr valign="top"><td><code>variables </code></td>
<td>
variables in the complete regression model</td></tr>
<tr valign="top"><td><code>G</code></td>
<td>
total number of input genes</td></tr>
<tr valign="top"><td><code>g</code></td>
<td>
number of genes taken in the regression fit</td></tr>
<tr valign="top"><td><code>dat </code></td>
<td>
input analysis data matrix</td></tr>
<tr valign="top"><td><code>dis</code></td>
<td>
regression design matrix</td></tr>
<tr valign="top"><td><code>step.method</code></td>
<td>
imputed step method for stepwise regression</td></tr>
<tr valign="top"><td><code>edesign</code></td>
<td>
matrix of experimental design</td></tr>
<tr valign="top"><td><code>influ.info </code></td>
<td>
data frame of genes containing influencial data</td></tr>
</table>

<h3>Author(s)</h3>

<p>
Ana Conesa, aconesa@ivia.es; María José Nueda, mj.nueda@ua.es
</p>


<h3>References</h3>

<p>
Conesa, A., Nueda M.J., Alberto Ferrer, A., Talón, T. 2005.
maSigPro: a Method to Identify Significant Differential Expression Profiles in Time-Course Microarray Experiments.
</p>


<h3>See Also</h3>

<p>
<code><a href="p.vector.html">p.vector</a></code>, <code><a onclick="findlink('stats', 'step.html')" style="text-decoration: underline; color: blue; cursor: hand">step</a></code>
</p>


<h3>Examples</h3>

<pre>

#### GENERATE TIME COURSE DATA
## generate n random gene expression profiles of a data set with 
## one control plus 3 treatments, 3 time points and r replicates per time point.

tc.GENE &lt;- function(n, r,
             var11 = 0.01, var12 = 0.01,var13 = 0.01,
             var21 = 0.01, var22 = 0.01, var23 =0.01,
             var31 = 0.01, var32 = 0.01, var33 = 0.01,
             var41 = 0.01, var42 = 0.01, var43 = 0.01,
             a1 = 0, a2 = 0, a3 = 0, a4 = 0,
             b1 = 0, b2 = 0, b3 = 0, b4 = 0,
             c1 = 0, c2 = 0, c3 = 0, c4 = 0)
{

  tc.dat &lt;- NULL
  for (i in 1:n) {
    Ctl &lt;- c(rnorm(r, a1, var11), rnorm(r, b1, var12), rnorm(r, c1, var13))  # Ctl group
    Tr1 &lt;- c(rnorm(r, a2, var21), rnorm(r, b2, var22), rnorm(r, c2, var23))  # Tr1 group
    Tr2 &lt;- c(rnorm(r, a3, var31), rnorm(r, b3, var32), rnorm(r, c3, var33))  # Tr2 group
    Tr3 &lt;- c(rnorm(r, a4, var41), rnorm(r, b4, var42), rnorm(r, c4, var43))  # Tr3 group
    gene &lt;- c(Ctl, Tr1, Tr2, Tr3)
    tc.dat &lt;- rbind(tc.dat, gene)
  }
  tc.dat
}

## Create 270 flat profiles
flat &lt;- tc.GENE(n = 270, r = 3)
## Create 10 genes with profile differences between Ctl and Tr1 groups
twodiff &lt;- tc.GENE (n = 10, r = 3, b2 = 0.5, c2 = 1.3)
## Create 10 genes with profile differences between Ctl, Tr2, and Tr3 groups
threediff &lt;- tc.GENE(n = 10, r = 3, b3 = 0.8, c3 = -1, a4 = -0.1, b4 = -0.8, c4 = -1.2)
## Create 10 genes with profile differences between Ctl and Tr2 and different variance
vardiff &lt;- tc.GENE(n = 10, r = 3, a3 = 0.7, b3 = 1, c3 = 1.2, var32 = 0.03, var33 = 0.03)
## Create dataset
tc.DATA &lt;- rbind(flat, twodiff, threediff, vardiff)
rownames(tc.DATA) &lt;- paste("feature", c(1:300), sep = "")
colnames(tc.DATA) &lt;- paste("Array", c(1:36), sep = "")
tc.DATA [sample(c(1:(300*36)), 300)] &lt;- NA  # introduce missing values

#### CREATE EXPERIMENTAL DESIGN
Time &lt;- rep(c(rep(c(1:3), each = 3)), 4)
Replicates &lt;- rep(c(1:12), each = 3)
Control &lt;- c(rep(1, 9), rep(0, 27))
Treat1 &lt;- c(rep(0, 9), rep(1, 9), rep(0, 18))
Treat2 &lt;- c(rep(0, 18), rep(1, 9), rep(0,9))
Treat3 &lt;- c(rep(0, 27), rep(1, 9))
edesign &lt;- cbind(Time, Replicates, Control, Treat1, Treat2, Treat3)
rownames(edesign) &lt;- paste("Array", c(1:36), sep = "")

## run T.fit from a p.vector object
tc.p &lt;- p.vector(tc.DATA, design = make.design.matrix(edesign), Q = 0.01) 
tc.tstep &lt;- T.fit(data = tc.p , alfa = 0.05)

## run T.fit from a data matrix and a design matrix
dise &lt;- make.design.matrix(edesign)
tc.tstep &lt;- T.fit (data = tc.DATA[271:300,], design = dise$dis, 
                   step.method = "two.ways.backward", min.obs = 10, alfa = 0.05)
tc.tstep$sol # gives the p.values of the significant 
             # regression coefficients of the optimized models
</pre>

<script Language="JScript">
function findlink(pkg, fn) {
var Y, link;
Y = location.href.lastIndexOf("\\") + 1;
link = location.href.substring(0, Y);
link = link + "../../" + pkg + "/chtml/" + pkg + ".chm::/" + fn;
location.href = link;
}
</script>


<hr><div align="center">[Package <em>maSigPro</em> version 1.0 <a href="00Index.html">Index]</a></div>

</body></html>
